src/thunderbots/software/test/world/robot.cpp

```diff
@@ -1,214 +1,328 @@
+    robot.updateState(Point(-1.2, 3), Vector(2.2, -0.05), Angle::quarter(),
+                      AngularVelocity::ofRadians(1.1), half_second_future);
 
     EXPECT_EQ(0, robot.id());
     EXPECT_EQ(Point(-1.2, 3), robot.position());
     EXPECT_EQ(Vector(2.2, -0.05), robot.velocity());
     EXPECT_EQ(Angle::quarter(), robot.orientation());
     EXPECT_EQ(AngularVelocity::ofRadians(1.1), robot.angularVelocity());
+    EXPECT_EQ(half_second_future, robot.lastUpdateTimestamp());
 }
 
-TEST(RobotTest, update_specific_params)
+TEST_F(RobotTest, update_state_with_specific_params)
 {
-    Robot robot = Robot(0);
+    Robot robot = Robot(0, current_time);
 
     // Only update certain parameters and leave the rest as their previous values
-    robot.update(robot.position(), Vector(2.2, -0.05), robot.orientation(),
-                 AngularVelocity::ofRadians(1.1));
+    robot.updateState(robot.position(), Vector(2.2, -0.05), robot.orientation(),
+                      AngularVelocity::ofRadians(1.1), current_time);
 
     EXPECT_EQ(0, robot.id());
     EXPECT_EQ(Point(), robot.position());
     EXPECT_EQ(Vector(2.2, -0.05), robot.velocity());
     EXPECT_EQ(Angle::zero(), robot.orientation());
     EXPECT_EQ(AngularVelocity::ofRadians(1.1), robot.angularVelocity());
+    EXPECT_EQ(current_time, robot.lastUpdateTimestamp());
+}
+
+TEST_F(RobotTest, update_state_with_specific_params_2)
+{
+    Robot robot = Robot(0, Point(), Vector(2.2, -0.05), Angle::zero(),
+                        AngularVelocity::ofRadians(1.1), current_time);
 
     // Repeat with a different combination of parameters
-    robot.update(Point(-1.2, 3), robot.velocity(), Angle::quarter(),
-                 robot.angularVelocity());
+    robot.updateState(Point(-1.2, 3), robot.velocity(), Angle::quarter(),
+                      robot.angularVelocity(), one_second_future);
 
     EXPECT_EQ(0, robot.id());
     EXPECT_EQ(Point(-1.2, 3), robot.position());
     EXPECT_EQ(Vector(2.2, -0.05), robot.velocity());
     EXPECT_EQ(Angle::quarter(), robot.orientation());
     EXPECT_EQ(AngularVelocity::ofRadians(1.1), robot.angularVelocity());
+    EXPECT_EQ(one_second_future, robot.lastUpdateTimestamp());
 }
 
-TEST(RobotTest, update_with_new_robot)
+TEST_F(RobotTest, update_state_with_new_robot_with_same_id)
 {
-    Robot robot = Robot(0);
+    Robot robot = Robot(0, current_time);
 
     Robot update_robot = Robot(0, Point(-1.2, 3), robot.velocity(), Angle::quarter(),
-                               robot.angularVelocity());
+                               robot.angularVelocity(), current_time);
 
-    robot.update(update_robot);
+    robot.updateState(update_robot);
 
     EXPECT_EQ(robot, update_robot);
+}
+
+TEST_F(RobotTest, update_with_new_robot_with_different_id)
+{
+    // TODO: Add unit test to check for a thrown exception when a robot with a different
+    // id is used once https://github.com/UBC-Thunderbots/Software/issues/16i is done
```

Comment:
Extra "i" at end of URL, ie `id is used once https://github.com/UBC-Thunderbots/Software/issues/16i is done` -> `id is used once https://github.com/UBC-Thunderbots/Software/issues/16 is done`

Reply:
fixed